src/ShallowWrapper.js

```diff
@@ -520,6 +521,29 @@ export default class ShallowWrapper {
   }
 
   /**
+   * Return a JSON representation of the node.
+   *
+   * NOTE: can only be called on a wrapper of a single node.
+   *
+   * @returns {Object}
+   */
+  json() {
+    return this.single('json', () => {
+      if (this.type() == null) {
+        return this.node;
+      }
+
+      const childrens = compact(this.children().map(c => c.json()));
+      return {
+        type: this.name(),
+        props: omit(this.props(), 'children'),
+        children: (childrens.length) ? childrens : null,
+        $$typeof: Symbol.for('react.test.json'),
+      };
```

Comment:
yeah, `Symbol.for` is an issue. `pretty-format` relies on it, though.

I agree now that the `json()` name can be confusing, I'm liking `snapshot()` more but usage will be a little weird: `expect(wrapper.snapshot()).toMatchSnapshot()` (or something similar)